build.gradle

```diff
@@ -28,32 +29,35 @@ dependencies {
 	implementation ('com.google.code.gson:gson:2.8.5')
 	implementation ('com.google.guava:guava:27.1-jre')
 
-	implementation ('net.fabricmc:sponge-mixin:0.9.1+mixin.0.8.2') {
+	implementation ('net.fabricmc:sponge-mixin:0.10.0+mixin.0.8.4.snapshot') {
 		exclude module: 'launchwrapper'
 		exclude module: 'guava'
 		exclude module: 'gson'
 	}
 }
 
 task sourcesJar(type: Jar, dependsOn: classes) {
-	classifier = 'sources'
+	archiveClassifier.set('sources')
 	from sourceSets.main.allSource
 }
 
 task javadocJar(type: Jar, dependsOn: javadoc) {
-	classifier = 'javadoc'
+	archiveClassifier.set('javadoc')
 	from javadoc.destinationDir
 }
 
 tasks.withType(JavaCompile).configureEach {
 	it.options.encoding = "UTF-8"
 
 	if (JavaVersion.current().isJava9Compatible()) {
-		it.options.release = 8
+		it.options.release.set(8)
```

Comment:
No, this isnt right, using = is a lot nicer imo, this also applies for the 2 cases above.